File: 프로그래머스/lv0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int idx = 0;
        int[] temp = new int[n];
        
        for(int i = 1; i <= n; i++) {
            if(i % 2 != 0) {
                temp[idx++] = i;
            }
        }
        int[] answer = new int[idx];
        for(int i = 0; i < idx; i++) {
            answer[i] = temp[i];
        }
        return answer;
    }
}

File: C/프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* answer = (int*)malloc(sizeof(int) * (n / 2 + 1));
    int index = 0;
    for(int i = 1; i <= n; i++)
    {
        if(i % 2 != 0) answer[index++] = i;
    }
    return answer;
}